fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -69,25 +70,13 @@ class ThresholdOptimizer(ClassifierMixin, BaseEstimator):
 
     def __init__(self, *, unconstrained_predictor=None, estimator=None,
                  constraints=DEMOGRAPHIC_PARITY, grid_size=1000, flip=True, plot=False):
-        if unconstrained_predictor and estimator:
-            raise ValueError(EITHER_PREDICTOR_OR_ESTIMATOR_ERROR_MESSAGE)
-        elif unconstrained_predictor:
-            self._unconstrained_predictor = unconstrained_predictor
-            self._estimator = None
-        elif estimator:
-            self._unconstrained_predictor = None
-            self._estimator = estimator
-        else:
-            raise ValueError(PREDICTOR_OR_ESTIMATOR_REQUIRED_ERROR_MESSAGE)
-
-        self._constraints = constraints
-        if self._constraints not in _SUPPORTED_CONSTRAINTS:
-            raise ValueError(NOT_SUPPORTED_CONSTRAINTS_ERROR_MESSAGE)
-
-        self._grid_size = grid_size
-        self._flip = flip
-        self._plot = plot
-        self._post_processed_predictor_by_sensitive_feature = None
+        self.grid_size = grid_size
+        self.flip = flip
+        self.plot = plot
```

Comment:
FYI plot is gone with my PR which I hope to complete today provided Richard/Miro manage to get a look

Reply:
... unless this goes in first of course

Reply:
Played around with this and managed to fix the flake8 issues. If the pipelines/tests pass I'll complete